base/range.jl

```diff
@@ -53,11 +53,20 @@ Given a starting value, construct a range either by length or from `start` to `s
 optionally with a given step (defaults to 1, a [`UnitRange`](@ref)).
 One of `length` or `stop` is required.  If `length`, `stop`, and `step` are all specified, they must agree.
 
-If `length` and `stop` are provided and `step` is not, the step size will be computed
-automatically such that there are `length` linearly spaced elements in the range (a [`LinRange`](@ref)).
```

Comment:
To me it would be clearer to keep the

> the step size will be computed automatically such that there are `length` linearly spaced elements in the range

part because it's important and applies to everything. Then say something like

```julia
"""
...in a range. The returned type depends on the inputs, a [`StepRangeLen`](@ref) for floating-point inputs and [`LinRange`](@ref) otherwise.
"""
``` 

because that's effectively a detail that most people won't need to worry about. (Since the docs for `StepRangeLen` mention error-free arithmetic, I think we're covered as far as "purpose" goes.)